frontend/src/actions/story.js

```diff
@@ -9,25 +9,54 @@ import {
   SUCCESS,
   FETCH_STORY_START,
   FETCH_STORY,
+  INVALID_STORY,
+  ERROR,
 } from "./types";
+import { fields } from "../routes/CriticalQuestions/fields";
+
+const mandatoryFields = [fields.CURRENT_LOCATION];
+
+const isValidStory = (dto) => {
+  return invalidFields(dto.story).length === 0;
+};
+
+const invalidFields = (dto) => {
+  return mandatoryFields.filter(
+    (field) => !(field.name in dto && dto[field.name])
+  );
+};
 
 export const submitStory = (dto) => async (dispatch) => {
-  dispatch({ type: SAVE_STORY_START });
-  const { story, nextPage } = dto;
-  const response = await api(`stories/`, {
-    method: "POST",
-    body: story,
-  });
+  if (isValidStory(dto)) {
```

Comment:
This is great! A style teaching tho (that I'm not sure I've mentioned before): _return early_. This premise says that if you're going to have an conditional that's using both of the logical branches, you pick the shortest-logic branch, put it first, and do a `return`. Then, the other branch is not unnecessarily indented.

This is trivial here, but when `if`s start nesting up it really helps to clear out the air!

Applying it here would make it look like:

```javascript
if (!isValidStory(dto)) 
   return dispatch({
      type: INVALID_STORY
      // ...etc, invalid story stuff
   })

// regular story saving stuff here
dispatch({ type: SAVE_STORY_START });
...
```